app/single_resource/views.py

```diff
@@ -150,15 +151,32 @@ def edit(resource_id):
                   'form-error')
     # Field id is not needed for the form, hence omitted with [1:].
     for field_name in resource_field_names[1:]:
-        form[field_name].data = resource.__dict__[field_name]
+        if form[field_name]:
+            form[field_name].data = resource.__dict__[field_name]
     return render_template('single_resource/edit.html',
                            form=form,
                            resource_id=resource_id)
 
 
-def save_associations(resource, form, descriptors, resource_existed=True):
+def save_associations(resource, form, descriptors, resource_existed):
     """Save associations from the forms received by 'create' and 'edit' route
     handlers to the database."""
+    #first delete all the associations for this resource if it already existed (to handle the "empty" case)
+    if resource_existed:
+        options = OptionAssociation.query.filter_by(resource_id = resource.id).all()
+        texts = TextAssociation.query.filter_by(resource_id = resource.id).all()
+        associations = options + texts
+        print "in this resource :)"
+        for a in associations:
+             print a
+             db.session.delete(a)
+             try:
```

Comment:
i think we should put this `try` block of committing outside of the for loop? so we try once to commit all deletions instead of trying each one?